### ISSUE TITLE
include lua with LuaSkin framework?

### ISSUE THREAD

**Issue by asmagill** [Hammerspoon/hammerspoon#2479]
@cmsj, an idea that I want to run by you...

When building the LuaSkin framework, would it be possible to build/include the following that are part of a normal lua installation?
* `include/lua.hpp` -- for c++ bindings
* `lib/liblua.a` -- the library external libraries built with `luarocks` require
* `bin/lua` -- lua interpreter
* `bin/luac` -- lua compiler -- I use this in Makefiles for testing lua files for syntax errors but it can also be used to create bytecode versions of lua files that may load quicker in some circumstances or provide a minimal layer of obfuscation, should someone desire it.

Given that you make changes to the Lua src to allow for the capturing of at least some objective-c exceptions, one or more of these may not be possible without also linking in some Frameworks...

But if they are, or if the additional linking is minimal, it would provide an easy way in the future to include (minus a config file I'll be describing in a wiki article to address #2473 later tonight) everything someone needs to build external modules from `luarocks` or other sources so that they would work with Hammerspoon without necessarily requiring them to also manually build the lua version we're running.

Just a thought...

**Comment by cmsj** [comment]
Yeah I think this makes a good amount of sense. If we build those things without `-DLUA_USE_OBJC_EXCEPTIONS` then they should be pretty vanilla.
